### PR TITLE
Add __str__ to Molecule

### LINKED ISSUE #34: Printing Molecule does not display anything
`print(Molecule())` does not display anything.

### PR DESCRIPTION
**Pull request by pckroon**
Fixes #34

### CODE REVIEW

File: vermouth/molecule.py [hunk 1]
```diff
@@ -352,6 +352,34 @@ def __eq__(self, other):
...
+        number_interactions = {'atoms': len(self.nodes)}
+
+        for interaction_type, interactions in self.interactions.items():
+            if not interactions:
+                continue
+            sort_keys[interaction_type] = len(interactions[0].atoms)
+            number_interactions[interaction_type] = len(interactions)
+
+        if number_interactions.get('bonds', 0) != len(self.edges):
+            number_interactions['edges'] = len(self.edges)
```
[the remark this answers is not in the document]
Comment (pckroon) [reply vermouth/molecule.py]: Because the edges mean something to us, but maybe not to all users. And the interesting case is when not every edge is a bond and vice versa.

[the remark this answers is not in the document]
Comment (jbarnoud) [reply vermouth/molecule.py]: OK on this one.

File: vermouth/tests/test_molecule.py [hunk 2]
```diff
@@ -1229,6 +1229,7 @@ def test_str_method(mol, moltype):
     found = str(mol)
     assert '{} {}'.format(len(mol), 'atoms') in found
     hypothesis.assume('' not in mol.interactions)
+    hypothesis.assume(all(itype not in '0123456789' for itype in mol.interactions))
```
Comment (jbarnoud) [inline vermouth/tests/test_molecule.py]: Why couldn't we have digits in interaction names?

Comment (pckroon) [reply vermouth/tests/test_molecule.py]: It was because I check `if itype not in found` if there are no interactions associated. And a valid `str(molecule)` *can* contain digits. However, correcting the test opened up a lot of edge cases where you have e.g. the following: `interactions = {'aa': [...], 'a': []}` causing failures.
Test has been adjusted to ensure `0 a` is not in found.
